python/sports-league-scheduling/tests/test_constraints.py

```diff
@@ -0,0 +1,165 @@
+              classic_match=False),
+    ]
+    # match with the second team
+    constraint_verifier.verify_that(start_to_away_hop).given(*matches).penalizes_by(5)
+
+
+def test_home_to_away_hop():
+    home_team = Team(id="1", name="TeamA", distance_to_team={"3": 7})
+    second_team = Team(id="2", name="TeamB", distance_to_team={})
+    third_team = Team(id="3", name="TeamC", distance_to_team={"1": 7})
+    matches = [
+        Match(id="1", home_team=home_team, away_team=second_team, round=Round(index=0, weekend_or_holiday=False),
+              classic_match=False),
+        Match(id="2", home_team=third_team, away_team=home_team, round=Round(index=1, weekend_or_holiday=False),
+              classic_match=False)
+    ]
+    # match with the home team
+    constraint_verifier.verify_that(home_to_away_hop).given(*matches).penalizes_by(7)
+
+
+def test_away_to_away_hop():
+    second_team = Team(id="2", name="TeamB", distance_to_team={"3": 2})
+    third_team = Team(id="3", name="TeamC", distance_to_team={"2": 2})
+    matches = [
+        Match(id="1", home_team=second_team, away_team=Team(id="1", name="TeamA", distance_to_team={}),
+              round=Round(index=0, weekend_or_holiday=False),classic_match=False),
+        Match(id="2", home_team=third_team, away_team=Team(id="1", name="TeamA", distance_to_team={}),
+              round=Round(index=1, weekend_or_holiday=False),classic_match=False)
+    ]
+    # match with the home team
+    constraint_verifier.verify_that(away_to_away_hop).given(*matches).penalizes_by(2)
+
+
+def test_away_to_home_hop():
+    home_team = Team(id="1",name="TeamA", distance_to_team={"2": 20})
+    second_team = Team(id="2", name="TeamB", distance_to_team={"1": 20})
+    matches = [
+        Match(id="1", home_team=second_team, away_team=home_team, round=Round(index=0, weekend_or_holiday=False),
+              classic_match=False),
+        Match(id="2", home_team=home_team, away_team=Team(id="3", name="TeamC", distance_to_team={}),
+              round=Round(index=1, weekend_or_holiday=False),classic_match=False)
+    ]
+    # match with the home team
+    constraint_verifier.verify_that(away_to_home_hop).given(*matches).penalizes_by(20)
+
+
+def test_away_to_end_hop():
+    home_team = Team(id="1", name="TeamA", distance_to_team={"3": 15})
+    third_team = Team(id="3", name="TeamC", distance_to_team={"1": 15})
+    matches = [
+        Match(id="1", home_team=home_team, away_team=Team(id="2", name="TeamB", distance_to_team={}),
+              round=Round(index=0, weekend_or_holiday=False),classic_match=False),
+        Match(id="2", home_team=third_team, away_team=home_team, round=Round(index=1, weekend_or_holiday=False),
+              classic_match=False)
+    ]
+    # match with the home team
+    constraint_verifier.verify_that(away_to_end_hop).given(*matches).penalizes_by(15)
+
+
+def test_classic_matches():
+    home_team = Team(id="1", name="TeamA", distance_to_team={})
```

Comment:
Don't need to specify `distance_to_team` if you are not populating it (it will create an empty dict using the default factory)

(applies to the whole file wherever `distance_to_team={}` is used).
```suggestion
    home_team = Team(id="1", name="TeamA")
```